services/QuillLMS/db/structure.sql

```diff
@@ -1612,12 +1612,13 @@ ALTER SEQUENCE public.comprehension_feedbacks_id_seq OWNED BY public.comprehensi
 
 CREATE TABLE public.comprehension_highlights (
     id integer NOT NULL,
-    feedback_id integer NOT NULL,
+    feedback_id integer,
```

Comment:
Hmm, so you are reusing the current `Evidence::Highlight` class for both, and some will have a `feedback_id` and some will have a `highlight_group_id`. If you go this route, you'd have to add conditional logic (`feedback` or `highlight_group`) to [most methods in the highlight class](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/engines/evidence/app/models/evidence/highlight.rb#L20-L110). So as part of this code, we'd need to make the highlight class methods still work in both setups (it'll be easy to write some failing tests for these).

Some paths you might consider:
1) Write one (or two) conditional methods that are used by other methods to point to the relevant object (feedback or rule or whatever)
2) Making the relationship polymorphic
3) A separate class for GenAI Highlights and copy the relevant code